tools/c7n_azure/c7n_azure/query.py

```diff
@@ -118,5 +118,11 @@ def augment(self, resources):
         #TODO: temporary put here. Applicable only to ARM resources. Need to move to ARMResourceManager base class
         for resource in resources:
             if 'id' in resource:
-                resource['resourceGroup'] = resource['id'].split('/')[4]
+                resource['resourceGroup'] = ResourceIdParser.get_resource_group(resource['id'])
         return resources
+
+    def get_resources(self, resource_ids):
+        resource_client = self.get_client('azure.mgmt.resource.ResourceManagementClient')
+        session = local_session(self.session_factory)
+        data = [resource_client.resources.get_by_id(rid, session.resource_api_version(rid)) for rid in resource_ids]
```

Comment:
I think this is a reasonable way to implement the base version of this but do we also need an overriden version to make this scenario work on the load balancer? I thought this just gives us back `GenericResource` objects without many fields.

Reply:
Yeah, I thought so, that’s why I went different route at the beginning. But it does work - at least in this case GenericResource's properties contains all the same fields as PublicIP. So for the cases when it doesn’t, we’ll need to overwrite.

Reply:
Well that seems OK then - if other scenarios on other resource types need more fields we can add resource specific implementations.